rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -199,10 +206,12 @@ where
                 }
 
                 // Update cursor
-                if let Err(err) = cursor.update(logs, range).await {
-                    warn!(?err, "Error updating cursor");
-                    break Some(SLEEP_DURATION);
-                };
+                if !error {
+                    if let Err(err) = cursor.update(logs, range).await {
+                        warn!(?err, "Error updating cursor");
+                        break Some(SLEEP_DURATION);
+                    };
+                }
```

Comment:
if you actually get an error, could add a `warn` that we're not even gonna update the cursor to retry, including the range. Also can you pls add `range` to the `warn` above as a drive-by (the `Error updating cursor` one)

Reply:
Added